### PR TITLE
Fix bug that test framework gets stuck when debuggee outputs many lines

### LINKED ISSUE #318: Test doesn't work if debuggee outputs many lines
**Your environment**

* `ruby -v`:  3.1dev
* `rdbg -v`: 1.2.3

**Describe the bug**

```ruby
require_relative '../support/test_case'

module DEBUGGER__
  class FooTest < TestCase
    def test_foo
      debug_code ' 1| 200.times{|i| p i}' do
        type 'c'
        assert_finish
      end
    end
  end
end
```

becomes timeout with remote tests.

**Expected behavior**

it should not fail.

### LINKED ISSUE #322: The output from the remote debuggee can't be checked in the current test framework
@ko1-san  pointed out that there is a problem such as the content of the title.

After #318 is solved(I created [the PR](https://github.com/ruby/debug/pull/321) to solve it), I'll fix this problem.

### PR DESCRIPTION
**Pull request by ono-max**
Fixes #318

I also defined `assert_debuggee_out` method to fix #322

### CODE REVIEW

File: test/support/utils.rb [hunk 1]
```diff
@@ -316,6 +298,7 @@ def check_error(error, test_info)
     def kill_remote_debuggee remote_info
       return unless remote_info
 
+      sleep 0.1 until remote_info.reader_thread.kill
```
Comment (ko1) [inline test/support/utils.rb:301]: `Thraed#kill` returns self. Why do you use `until`?

File: test/support/assertions.rb [hunk 2]
```diff
@@ -57,6 +57,21 @@ def assert_no_line_text(text)
       })
     end
 
+    def assert_debuggee_out exp
+      @scenario.push(Proc.new {|test_info|
+        if test_info.mode == 'LOCAL'
+          log = test_info.last_backlog.join
+        else
+          log = test_info.remote_info.debuggee_backlog.join
```
Comment (ko1) [inline test/support/assertions.rb:65]: do you need `last_debuggee_backlog`?

File: test/support/assertions.rb [hunk 3]
```diff
@@ -57,6 +57,21 @@ def assert_no_line_text(text)
       })
     end
 
+    def assert_debuggee_out exp
```
Comment (ko1) [inline test/support/assertions.rb:60]: I think it should be consistent with `assert_line_text` (name and parameters)

Comment (ono-max) [reply test/support/assertions.rb:60]: https://github.com/ruby/debug/pull/365

Comment (ko1) [reply test/support/assertions.rb:60]: Just now it is not renamed yet, so it should be renamed after #308.
Or merge this pull request after #308?

Comment (ono-max) [reply test/support/assertions.rb:60]: Either is fine, but if I had to say, I prefer the former. To make it consistent with #365, we may need to change `assert_debuggee_out`, too.

Comment (ko1) [reply test/support/assertions.rb:60]: Former option means introduce current consistent name (maybe `assert_debugee_line_text` or similar) and rename it with #365. Could you try it?

Comment (ono-max) [reply test/support/assertions.rb]: I renamed it to "assert_debuggee_line_text".

**Pull request merged**